File: android/src/main/java/com/netrareactnative/dto/ResponseDTO.kt
package com.example.netra_flutter.dto

import com.netra.library.NetraResponse

data class ResponseDTO(
    val data: Any?,
    val statusCode: Int,
    val statusMessage: String?,
    val isCache: Boolean?,
    val headers: Map<String, String>?,
) {
    companion object {
        fun<T> fromDataModel(response: NetraResponse<T>): ResponseDTO {
            return ResponseDTO(
                data = response.data,
                isCache = response.isCache,
                statusCode = response.statusCode,
                statusMessage = response.statusMessage,
                headers = response.headers,
            )
        }
    }

    fun toDataModel(): NetraResponse<Any?> {
        return NetraResponse(
            data = data,
            isCache = isCache,
            statusCode = statusCode,
            statusMessage = statusMessage,
            headers = headers,
        )
    }
}

File: src/internal/dto/ResponseDTO.ts
import { Response } from '../../models/Response';
import { BaseDTO } from './BaseDTO';

export class ResponseDTO extends BaseDTO {
  data?: unknown;
  statusCode?: number;
  statusMessage?: string;
  headers?: Record<string, string>;

  constructor(
    data?: unknown,
    statusCode?: number,
    statusMessage?: string,
    headers?: Record<string, string>
  ) {
    super();
    this.data = data;
    this.statusCode = statusCode;
    this.statusMessage = statusMessage;
    this.headers = headers;
  }

  static fromDataModel<T>(model: Response<T>): ResponseDTO {
    return new ResponseDTO(
      model.data,
      model.statusCode,
      model.statusMessage,
      model.headers
    );
  }

  static fromJSON(json: string) {
    const parsedJSON = JSON.parse(json);
    return new ResponseDTO(
      parsedJSON.data,
      parsedJSON.statusCode,
      parsedJSON.statusMessage,
      parsedJSON.headers
    );
  }

  toDataModel<T>(): Response<T> {
    return Response.fromRaw<T>({
      data: this.data,
      statusCode: this.statusCode,
      statusMessage: this.statusMessage,
      headers: this.headers,
    });
  }

  toJSON() {
    return {
      data: this.data,
      statusCode: this.statusCode,
      statusMessage: this.statusMessage,
      headers: this.headers,
    };
  }
}
